pkg/pod/entrypoint_lookup_impl.go

```diff
@@ -67,7 +67,13 @@ func (e *entrypointCache) Get(ref name.Reference, namespace, serviceAccountName
 		return nil, fmt.Errorf("error creating k8schain: %v", err)
 	}
 	mkc := authn.NewMultiKeychain(kc)
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc))
+	//By default go-containerregistry pulls amd64 images.
+	//Setting correct image pull architecture based on the underlying platform
+	var pf = v1.Platform {
+        		Architecture: runtime.GOARCH ,
+	        	OS :          "linux",
```

Comment:
Perhaps for good measure this should also be `runtime.GOOS`?

Reply:
Good suggestion. Made the required changes.